src/client/components/line-chart-axis/line-chart-axis.tsx

```diff
@@ -20,58 +20,78 @@ import * as d3 from "d3";
 import * as moment from "moment-timezone";
 import * as React from "react";
 import { Stage } from "../../../common/models/stage/stage";
+import { Unary } from "../../../common/utils/functional/functional";
 import { roundToHalfPx } from "../../utils/dom/dom";
 import "./line-chart-axis.scss";
 
 const TICK_HEIGHT = 5;
 const TEXT_OFFSET = 12;
-const floatFormat = d3.format(".1f");
+
+type AxisScale = d3.time.Scale<Date, number> | d3.time.Scale<number, number>;
```

Comment:
For consideration.
We could **probably** extract AxisTick or Tick type = Date | number, and then use it here as d3.time.Scale<Tick, number> and in all following lines where "Date | number" is referenced.

I haven't checked it but should be possible in TypeScript.

Reply:
But we don't want to have scale that accepts Date or number. We want scale for Date or scale for number.